### PR TITLE
Add a regression test for #104322

### PR DESCRIPTION
**Pull request by weiznich**
r? @compiler-errors

### CODE REVIEW

File: src/test/ui/traits/issue-104322.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
+// run-pass
+//
+// Tests that overflows do not occure in certain situations
```
Comment (Rageking8) [inline src/test/ui/traits/issue-104322.rs:3]:
```suggestion
// Tests that overflows do not occur in certain situations
```

File: src/test/ui/traits/issue-104322.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
+// run-pass
```
Comment (compiler-errors) [inline src/test/ui/traits/issue-104322.rs:1]: This does not need to be run-pass, just build-pass.

**Review by compiler-errors — COMMENTED** [review]
Please amend the test and squash the commits

File: src/test/ui/traits/issue-104322.rs (hunk 2 above)
Comment (weiznich) [reply src/test/ui/traits/issue-104322.rs:1]:
```suggestion
// build-pass
```

**Pull request merged**